### ISSUE TITLE
Is there a way to know a model is "tool compatible" ?

### ISSUE THREAD

**Issue by LilPiep** [awaescher/OllamaSharp#74]

**Comment by LilPiep** [comment]
By that I mean is there a method to know, when deploying a model, if it's compatible with a Semantic Kernel usage. 
It looks like model.details doesn't have this information
Thanks for your help

**Comment by awaescher** [comment]
It seems Ollama has the meta data anywhere, as they are tagging their model cards with it:

![image](https://github.com/user-attachments/assets/0b5b4629-8b3c-4eb6-a3fd-74eb59af5107)

However, I don't know of anything in the "local" meta data after the manifest was pulled.

OllamaSharp will [throw an strongly typed `ModelDoesNotSupportToolsException`](https://github.com/awaescher/OllamaSharp/blob/4dd9b783ac8bbd458c0b1639b93bc268648889db/src/OllamaApiClient.cs#L305C42-L305C52) once it finds out that tools are not supported. But I guess that's too late at runtime, right?

**Comment by LilPiep** [comment]
Yep indeed. I would love to collect this data as a boolean to forbid the use of Semantic Kernel in order to avoid Errors.
But once the manifest is pulled, it seems like this information is lost

 
<img width="350" alt="image" src="https://github.com/user-attachments/assets/ef1ec4bf-1e87-4352-a8fe-a9a0c4fa92bb">

**Comment by LilPiep** [comment]
Thanks for you answer tho ! I will find another way to collect this data :)

**Comment by awaescher** [comment]
Maybe looking in the Ollama repository how they find out.

**Comment by awaescher** [comment]
[Maybe here](https://github.com/ollama/ollama/blob/fe91d7fff13cc48879b320911a9662f08a686264/server/images.go#L103)
